library/ssl_tls.c

```diff
@@ -1615,11 +1620,75 @@ int mbedtls_ssl_set_hs_ecjpake_password( mbedtls_ssl_context *ssl,
                                          const unsigned char *pw,
                                          size_t pw_len )
 {
```

Comment:
Thinking while reading this: we probably want to also add a new function `mbedtls_ssl_set_hs_ecjpake_password_opaque()` that takes the password directly from a key slot, similarly to how we added `mbedtls_ssl_conf_psk_opaque()` to complement `mbedtls_ssl_conf_psk()`.

The reasoning is, the EC J-PAKE password is presumably a long-term secret, so it should be possible to hold it on the PSA side (where it can benefit from better isolation), just like all other long-term secret. The current API requires it to be accessible in clear to the TLS implementation.

I'll create a new issue about that, in order to avoid scope creep for this PR.

Reply:
Issue created: https://github.com/Mbed-TLS/mbedtls/issues/6599

Reply:
Ok, I see. Once this is closed I can take a look to that one as well